test/rules/do_not_catch_errors.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N do_not_catch_errors`
+
+void bad() {
+  try {} on A catch (e) { // LINT
+    // ignore
+  } on B catch (e) { // LINT
```

Comment:
Not exactly what I suggested: you added some extra text. :-) The point of the test case I gave is that in Dart the `catch` clause is optional if you have an `on` clause, and if the `on` clause specifies a subclass of `Error` then I think you still want to produce a lint.

(Having a test case with a class that indirectly inherits from `Error` is a good enhancement.)